# Titanic 


In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
#pd.options.mode.chained_assignment = None # disable chained assignment warning
import seaborn as sns
from scipy.stats import norm

In [2]:

from datetime import datetime
datetime.today().timetuple().tm_yday

285

## Explore the Titanic Data

In [3]:
titanic = sns.load_dataset('titanic')
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [4]:
titanic[['survived']].mean()

survived    0.383838
dtype: float64

### Task 

We have seen a strong dependence of the *outcome* on the two "variables"/"features"/"regressors" *pclass* and *sex*.
The natural question is whether there could be more factors "correlated with"/"influencing"/"affecting" Survival.


1. Does the port of embarkment matter ?
    * (**MC**) What is the distribution (counts) of embarkment? (Hint: look at `pd.value_counts` )
        * **A** 168, 77, 644
        * **B** 158, 80, 636
        * **C** 170, 75, 639
        * **D** 164, 79, 667
    * (**MC**) What are the survival rates for *Southampton* as a function of `pclass`?
        * **A** 0.54, 0.42, 0.17
        * **B** 0.62, 0.39, 0.15
        * **C** 0.58, 0.46, 0.19
        * **D** 0.56, 0.37, 0.21
    * Do the survival rates "look" different from *Cherbourg* ?
    * How would you make sure that the observed differences are not due to chance ?
    
2. Does the *fare* paid matter ?
    * How would you quantify/visualize this ? 
    * What is the fundamental difference between the previous relationship of two variables ?
    * Have you heard of the terms *confounding* or *confounders* or *marginal dependence* versus *conditional dependence* ? 
    * Discuss dependencies among the features. Revisit the port of embarkment question in this light !
3. Does *age* matter ?
    * (**MC**) What is the survival rate for passengers below the age of 18?
        * **A** 0.47
        * **B** 0.74
        * **C** 0.54
        * **D** 0.45
    * (**MC**) What are the survival rates for passengers below the age of 18 stratified by pclass?
        * **A** 0.91, 0.87, 0.36
        * **B** 0.93, 0.88, 0.38
        * **C** 0.95, 0.93, 0.37
        * **D** 0.92, 0.91, 0.37
    * How would you make sure that the observed differences are not due to chance ?
    
